File: core/src/remove/tanks/game/level/utility/stage/broker/message/MessagePrefab.java
package remove.tanks.game.level.utility.stage.broker.message;

import com.google.common.base.MoreObjects;
import remove.tanks.game.utility.Prefab;

/**
 * @author Mateusz Długosz
 */
public final class MessagePrefab extends Prefab {
    private final float time;
    private final String entryKey;
    private final String faceId;

    public MessagePrefab(float time, String entryKey, String faceId) {
        this.time = time;
        this.entryKey = entryKey;
        this.faceId = faceId;
    }

    public float getTime() {
        return time;
    }

    public String getEntryKey() {
        return entryKey;
    }

    public String getFaceId() {
        return faceId;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this)
                .add("time", time)
                .add("entryKey", entryKey)
                .add("faceId", faceId)
                .toString();
    }
}

File: tools/docs-generator/lib/level/utility/stage/broker/message/message_prefab.py
class MessagePrefab(object):
    def __init__(self, time, translation_entry_key, face_id):
        self.time = float(time)
        self.translation_entry_key = str(translation_entry_key)
        self.face_id = str(face_id)

    def get_time(self):
        return self.time

    def get_translation_entry_key(self):
        return self.translation_entry_key

    def get_face_id(self):
        return self.face_id

    def __str__(self):
        return "MessagePrefab(time={}, translation_entry_key={}, face_id={})" \
            .format(self.time, self.translation_entry_key, self.face_id)
